training-1/src/main.rs

```diff
@@ -126,8 +131,11 @@ fn main() {
         }
 
         if !level_complete {
-            if let Some(cmd) = vm.console.try_process_command() {
-                vm.execute_command(cmd);
+            if let Some(cmd) = vm.console.get_next_command() {
```

Comment:
I think the borrow checker allows us having a struct which wraps the virtual machine, the console and the command processor. I'll look into this.

Another idea I had is to move all the boilerplate involved in setting up the window, initializing render stuff, etc. into a separate function, so each level is only a minimal amount of code. However, to get this to work, I don't think we can have the level initialize a large struct which takes care of all the logic, because the borrow checker would not allow it. We would need something like this:

```rust
fn main() {
    launch_game(custom_update, custom_render);
}

fn custom_update(...) { ... }
fn custom_render(...) { ... }
```

Here, `launch_game` would take care of all the stuff both levels are currently doing in their main function, in addition to handling command processing, etc. Ideally, this would allow us to separate the virtual machine from a lot of the components it depends on. Do you think this is a viable way of going about things?